pkg/auth/auth.go

```diff
@@ -34,8 +35,20 @@ func CheckAuthFile(authfile string) error {
 }
 
 // Login login to the server with creds from Stdin or CLI
-func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginOptions, registry string) error {
-	server := getRegistryName(registry)
+func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginOptions, registry []string) error {
+	var server string
+	if len(registry) == 0 {
+		registriesFromFile, err := registries.GetRegistries()
+		if err != nil || len(registriesFromFile) == 0 {
+			return errors.Errorf("please specify a registry to login to")
```

Comment:
If there is an error, wouldn’t it be better to report it?

That implies a bit more logic, so maybe most of the new code could be factored out into a shared helper called by both the login/logout cases.